extensions/ql-vscode/src/remote-queries/remote-queries-manager.ts

```diff
@@ -301,4 +302,59 @@ export class RemoteQueriesManager extends DisposableObject {
       queryItem.status = QueryStatus.Failed;
     }
   }
+
+  public async exportVariantAnalysisResults(): Promise<void> {
```

Comment:
A suggestion (that is definitely not necessary) around how this bit of code could be made a bit cleaner (and btw I think what you have is fine):
- Move the `exportVariantAnalysisResults` function to another file/module and pass in any dependencies (e.g. `logger`, `qhm`). This can be done as a class that receives the dependencies in the constructor (like we do in the `RemoteQueriesManager`), or it could be simply be a function within a `results-sharing.ts` module. 
- Once we're in a new file, it'd be easier to break down into smaller ones a bit. E.g. The part around selecting which mode to use (gist or local) can be in one function, and the bits around generating gists or local files could be in separate functions.
 
Totally up to you if you want to do any of that. I just thought it might be useful to write down.


Also while writing this I realise that we've generally used the term remote query rather than variant analysis in non user-facing stuff. I'd personally try and stick to that to have some sort of consistency (although I appreciate it's by definition inconsistent 😬 )